File: backends/php-laravel/app/Domain/Client/ValueObjects/JwtHistoryListVo.php
<?php

/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */

declare(strict_types=1);

namespace App\Domain\Client\ValueObjects;

use App\Support\Enums\SortType;
use App\Support\Traits\Getter;
use App\Support\ValueObjects\AbstractValueObject;
use Illuminate\Support\Collection;

/**
 * JWT履歴一覧ValueObjectクラスです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 * @package App\Domain\Client\ValueObjects
 */
class JwtHistoryListVo extends AbstractValueObject
{
    use Getter;

    private Collection $items;
    private int $count = 0;
    private int $offset = 0;
    private int $limit = 0;
    private string $sort = '';
    private SortType $sortType = SortType::NONE;

    /**
     * {@inheritdoc}
     */
    #[\Override]
    public function attributes(): array
    {
        return [
            'items'    => $this->items->all(),
            'count'    => $this->count,
            'offset'   => $this->offset,
            'limit'    => $this->limit,
            'sort'     => $this->sort,
            'sortType' => $this->sortType,
        ];
    }

    /**
     * エンティティコレクションからアイテムを設定します。
     *
     * @param Collection $list エンティティのコレクション
     */
    public function assignItems(Collection $list): void
    {
        foreach ($list as $entity) {
            $this->items->add($entity->attributesBySnake());
        }
    }

    /**
     * 総件数を設定します。
     *
     * @param int $count 総件数
     */
    public function setCount(int $count): void
    {
        $this->count = $count;
    }

    /**
     * ページング情報を設定します。
     *
     * @param int $offset オフセット
     * @param int $limit 取得件数
     * @param string $sort ソート対象
     * @param SortType $sortType ソート順
     */
    public function setPaging(int $offset, int $limit, string $sort, SortType $sortType): void
    {
        $this->offset = $offset;
        $this->limit = $limit;
        $this->sort = $sort;
        $this->sortType = $sortType;
    }
}

File: backends/java-spring/src/main/java/com/authorization/domain/client/valueobjects/JwtHistoryListVo.java
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
package com.authorization.domain.client.valueobjects;

import com.authorization.domain.client.entities.JwtHistory;
import com.authorization.support.enums.SortType;
import com.authorization.support.valueobjects.AbstractValueObject;
import java.util.List;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * JWT履歴一覧ValueObjectクラスです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class JwtHistoryListVo extends AbstractValueObject {

    private List<JwtHistory> items = List.of();
    private int count;
    private int offset;
    private int limit;
    private String sort = "";
    private SortType sortType = SortType.NONE;
}
